src/sentry/static/sentry/app/views/organizationDiscover/queryBuilder.jsx

```diff
@@ -205,6 +208,37 @@ export default function createQueryBuilder(initial = {}, organization) {
       });
   }
 
+  /**
+   * Fetches either the query provided as an argument or the current query state
+   * if this is not provided and returns the result wrapped in a promise
+   *
+   * @param {Object} [data] Optional field to provide data to fetch
+   * @returns {Promise<Object|Error>}
+   */
+  function fetchForWidget(data = getExternal()) {
```

Comment:
Could be called `fetchWithoutLimit` and `API_LIMIT` vs `WIDGET_LIMIT ` since querybuilder shouldn't have to know about other UI components.